relevanceai/operations_new/sentiment/transform.py

```diff
@@ -98,60 +98,53 @@ def label_mapping(self):
 
     def analyze_sentiment(
         self,
-        text,
+        texts: List[str],
         highlight: bool = False,
         positive_sentiment_name: str = "positive",
         max_number_of_shap_documents: Optional[int] = None,
         min_abs_score: float = 0.1,
+        eps: float = 1e-9,
     ):
-        if text is None:
-            return None
-        labels = self.classifier([str(text)], truncation=True, max_length=512)
-        ind_max = np.argmax([l["score"] for l in labels[0]])
-        sentiment = labels[0][ind_max]["label"]
-        max_score = labels[0][ind_max]["score"]
-        sentiment = self.label_mapping.get(sentiment, sentiment)
-        if sentiment.lower() == "neutral" and max_score > self.sensitivity:
-            overall_sentiment = 1e-5
-        elif sentiment.lower() == "neutral":
-            # get the next highest score
-            new_labels = labels[0][:ind_max] + labels[0][(ind_max + 1) :]
-            new_ind_max = np.argmax([l["score"] for l in new_labels])
-            new_max_score = new_labels[new_ind_max]["score"]
-            new_sentiment = new_labels[new_ind_max]["label"]
-            new_sentiment = self.label_mapping.get(new_sentiment, new_sentiment)
-            overall_sentiment = self._calculate_overall_sentiment(
-                new_max_score, new_sentiment
-            )
+        labels = self.classifier(texts, truncation=True, max_length=512)
+        scores = np.array([[l["score"] for l in output] for output in labels])
+        e_x = scores[:, 0] * -1 + scores[:, 1] * 0 + scores[:, 2] * 1
```

Comment:
This assumes `negative` is always the first thing that is returned. Can you check this is true for all models? I don't think this is the case

Reply:
we have a hardcoded label mapping tho